File: problem-test/src/main/java/com/dudgns/problemtest/entity/carManagement/CarEntity.java
package com.dudgns.problemtest.entity.carManagement;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Builder
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "car")
public class CarEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idx;

    @OneToMany(fetch = FetchType.LAZY)
    @JoinColumn(name = "car_idx", insertable = false, updatable = false)
    private List<CarCategoryMappingEntity> carCategoryMappingEntityList = new ArrayList<>();

    @Column(name = "company_code", insertable = true, updatable = true)
    private Long companyCode;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_code", referencedColumnName = "company_code", insertable = false, updatable = false)
    private CompanyEntity companyEntity;

    @Column(name = "model_name")
    private String modelName;

    @Column(name = "created_year")
    private Integer createdYear;

    @Column(name = "rental_yn")
    private Boolean rentalYn;
}

File: problem-test/commons/project-dudgns-data-entity/src/main/kotlin/dudgns/com/backend/commons/data/entity/problemTest/CarEntity.kt
package dudgns.com.backend.commons.data.entity.problemTest

import jakarta.persistence.*

@Entity
@Table(name = "car")
class CarEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var idx: Long? = null

    @OneToMany(fetch = FetchType.LAZY)
    @JoinColumn(name = "car_idx", insertable = false, updatable = false)
    val carCategoryMappingEntityList: List<CarCategoryMappingEntity> = ArrayList<CarCategoryMappingEntity>()

    @Column(name = "company_code", insertable = true, updatable = true)
    var companyCode: Long? = null

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_code", referencedColumnName = "company_code", insertable = false, updatable = false)
    var companyEntity: CompanyEntity? = null

    @Column(name = "model_name")
    var modelName: String? = null

    @Column(name = "created_year")
    var createdYear: Int? = null

    @Column(name = "rental_yn")
    var rentalYn: Boolean? = null

    constructor(
        idx: Long?,
        companyCode: Long,
        modelName: String,
        createdYear: Int,
        rentalYn: Boolean
    ) {
        this.idx = idx
        this.companyCode = companyCode
        this.modelName = modelName
        this.createdYear = createdYear
        this.rentalYn = rentalYn
    }
}